pkg/cmd/cli/cmd/version.go

```diff
@@ -0,0 +1,183 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	etcdversion "github.com/coreos/etcd/version"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	kubeversion "k8s.io/kubernetes/pkg/version"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+	"github.com/openshift/origin/pkg/version"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	versionLong = `
+Display client and server versions.`
+)
+
+type VersionOptions struct {
+	BaseName         string
+	Config           clientcmdapi.Config
+	ClientConfig     *restclient.Config
```

Comment:
Looks like you'd only use this for a hostname?  Can be set without a remote call in `Complete` right?